RELEASE/scripts/autoscend/quests/level_08.ash

```diff
@@ -499,165 +469,266 @@ void theeXtremeSlopeChoiceHandler(int choice) {
+			if(pulls_remaining() == 0 && item_amount(it) == 0)
+			{
+				return false;	//out of pulls in softcore. come back tomorrow
+			}
 		}
 	}
 
-	if((item_amount($item[Ninja Rope]) >= 1) && (item_amount($item[Ninja Carabiner]) >= 1) && (item_amount($item[Ninja Crampons]) >= 1))
-	{
-		return false;
-	}
-
 	if(get_property("_sourceTerminalDigitizeMonster") == $monster[Ninja Snowman Assassin])
 	{
 		if(loopHandler("_auto_digitizeAssassinTurn", "_auto_digitizeAssassinCounter", "Potentially unable to do anything while waiting on digitized Ninja Snowman Assassin.", 10))
 		{
-			auto_log_info("Have a digitized Ninja Snowman Assassin, let's put off the Ninja Snowman Lair", "blue");
+			auto_log_info("Have a digitized Ninja Snowman Assassin, let's put off the Ninja Snowmen Lair", "blue");
 		}
 		return false;
 	}
 
-	if (in_hardcore()) {
-		if((have_effect($effect[Thrice-Cursed]) > 0) || (have_effect($effect[Twice-Cursed]) > 0) || (have_effect($effect[Once-Cursed]) > 0))
-		{
-			return false;
-		}
+	if((have_effect($effect[Thrice-Cursed]) > 0) || (have_effect($effect[Twice-Cursed]) > 0) || (have_effect($effect[Once-Cursed]) > 0))
+	{
+		return false;
+	}
 
-		if (shenShouldDelayZone($location[Lair of the Ninja Snowmen]))
-		{
-			auto_log_debug("Delaying Lair of the Ninja Snowmen in case of Shen.");
-			return false;
-		}
+	if (shenShouldDelayZone($location[Lair of the Ninja Snowmen]))
+	{
+		auto_log_debug("Delaying Lair of the Ninja Snowmen in case of Shen.");
+		return false;
+	}
 
-		if (isActuallyEd() && !elementalPlanes_access($element[spooky]))
+	//can we provide enough combat bonus to make it worthwhile trying to get snowman assassins?
+	if(providePlusCombat(25, true, true) <= 0.0)	//ninja snowman does not show up if +combat is not greater than 0
+	{
+		if(isAboutToPowerlevel())
 		{
-			adjustEdHat("myst");
-		}
-
-		if (providePlusCombat(25, true, true) <= 0.0) {
-			auto_log_warning("Something is keeping us from getting a suitable combat rate, we have: " + numeric_modifier("Combat Rate") + " and Ninja Snowmen.", "red");
-			return false;
-		}
-
-		if (autoAdv($location[Lair of the Ninja Snowmen])) {
+			auto_log_info("Something is keeping us from getting a suitable combat rate for ninja snowman assassin. we can only reach: " + numeric_modifier("Combat Rate") + ". Since we are about to powerlevel we are giving up on ninja route", "red");
+			set_property("auto_L8_ninjaSkip", true);		//give up on ninja route for this ascension
 			return true;
 		}
-		auto_log_warning("Seems like we failed the Ninja Snowmen unlock, reverting trapper setting", "red");
+		else
+		{
+			auto_log_warning("Something is keeping us from getting a suitable combat rate for ninja snowman assassin. we can only reach: " + numeric_modifier("Combat Rate") + ". Will delay and try again later", "red");
+		}
+		return false;
+	}
+	
+	//buff
+	if (isActuallyEd() && !elementalPlanes_access($element[spooky]))
+	{
+		adjustEdHat("myst");
+	}
+	providePlusCombat(25, true);
```

Comment:
also already handled in `acquireCombatMods()`/`zone_combatMod()`

Reply:
thank you. fixed